softwareComponents/atoms/include/atoms/cmdline_utils.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <concepts>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+
+
+namespace atoms
+{
+
+/**
+ * @brief Reads input from source \p inputFilePath .
+ * If the value of \p inputFilePath is `-`, `std::cin` is used as the source.
+ * Otherwise the value is treated as a file name.
+ * @param inputFilePath path to file containing the input
+ * @throws if the file \p inputFilePath could not be opened
+ * @returns the result of invocation of `readCallback`
+ */
+auto readInput( const std::filesystem::path & inputFilePath,
```

Comment:
Ah, jen si postesknu nad C++, že je škoda, že ten stream prostě nejde vrátit :/

Reply:
Jakoby, šlo by to se strukturou, kde si držím `unique_ptr< ifstream >` a přetížím operator>> aby to bral buď z ifstream anebo z std::cin... (případně přes variant)

Asi by se dala ta funkce pojmenovat jako `open` a tím dát najevo, že to drží otevřený zdroj...

Takže jakoby šlo by to, jenom takhle to bylo jednodušší

Reply:
Jo, souhlasím s tím, jak jsi to udělal. Jen jsem si chtěl postěžovat, že to jednoduše nejde (přitom bych zrovna ocenil, aby knihovna nabízela proxy wrapper pro `cin` a `cout`).